lib/oauth2/mac_token.rb

```diff
@@ -7,6 +7,19 @@
 
 module OAuth2
   class MACToken < AccessToken
+    # Initializes an MACToken from a Hash
+    #
+    # @param client [Client] the OAuth2::Client instance
+    # @param hash [Hash] a hash of MACToken property values
+    # @return [MACToken] the initialized AccessToken
+    def self.from_hash(client, hash)
+      hash = hash.dup
+      new(client,
+          hash.delete('access_token') || hash.delete(:access_token),
+          hash.delete('secret') || hash.delete(:secret),
```

Comment:
I think we need to handle `mac_key` and `mac_algorithm` according to https://tools.ietf.org/id/draft-ietf-oauth-v2-http-mac-02.html.